test/e2e/network/traffic_policy.go

```diff
@@ -0,0 +1,453 @@
+	f := framework.NewDefaultFramework("traffic-policy")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	// filled in BeforeEach
+	var c clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		e2eskipper.SkipUnlessMultizone(c)
+		e2eskipper.SkipUnlessNodeCountIsAtLeast(3)
+	})
+
+	ginkgo.It("honors hints when ITP and ETP use Cluster policies", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeCluster, v1.ServiceInternalTrafficPolicyCluster)
+
+		ginkgo.By("using the endpoint zone for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawMoreThanOnePod)
+
+		ginkgo.By("using the endpoint zone for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, tr.endpointZone, tr.endpointNode.Name, sawMoreThanOnePod)
+	})
+
+	ginkgo.It("honors hints only for internal traffic when ETP is local", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeLocal, v1.ServiceInternalTrafficPolicyCluster)
+
+		ginkgo.By("using the endpoint node for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawOnePodOnEndpointNode)
+
+		ginkgo.By("using the endpoint zone for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, tr.endpointZone, tr.endpointNode.Name, sawMoreThanOnePod)
+	})
+
+	ginkgo.It("honors hints only for external requests when ITP is local", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeCluster, v1.ServiceInternalTrafficPolicyLocal)
+
+		ginkgo.By("using the endpoint zone for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawMoreThanOnePod)
+
+		ginkgo.By("using the endpoint node for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, "", tr.endpointNode.Name, sawOnePodOnEndpointNode)
+	})
+
+	ginkgo.It("ignores hints when ITP and ETP are both local", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeLocal, v1.ServiceInternalTrafficPolicyLocal)
+
+		ginkgo.By("using the endpoint node for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawOnePodOnEndpointNode)
+
+		ginkgo.By("using the endpoint node for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, "", tr.endpointNode.Name, sawOnePodOnEndpointNode)
+	})
+})
+
+var _ = common.SIGDescribe("Topology Hints Disbled[Feature:Topology Hints][Feature:ServiceInternalTrafficPolicy]", func() {
+	f := framework.NewDefaultFramework("traffic-policy")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	// filled in BeforeEach
+	var c clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		e2eskipper.SkipUnlessMultizone(c)
```

Comment:
does this need multizone?

Reply:
I think so. I wanted to use multizone here to be sure that topology hints are truly ignored.

Reply:
I got it now, thanks